### PR TITLE
Fix the alignment issue in RTL

### LINKED ISSUE #20803: [Site selection screen]: Site name and url overflows in RTL orientation
### Actual behaviour 
Site name and url overflows in RTL orientation 

### Steps to reproduce the behavior
1. Go to Me → App language 
2. Set language as Arabic or any other RTL language 

Screenshot 2024-05-13 at 11 22 01 AM

##### Tested on [device], Android [version], JPAndroid / WPAndroid [version]
Pixel 6a, Android 13, [`trunk`] branch

### PR DESCRIPTION
**Pull request by jarvislin**
Fixes #20803 

The RTL in the screen of Site Picker was not handled during the development so this PR fixes the alignment issue in RTL mode.

| Before | After |
|--|--|
|  | Screenshot_٢٠٢٤٠٥١٣-١١١٠١٥ |

-----

## To Test:

1. Sign in JP app
2. Go to `Me` tab → App language
3. Set language as Arabic or any other RTL language
4. Go to `My Site` tab
5. Click the `v` arrow icon at the top of screen to open Site Picker
6. The UI in RTL should be correct.
7. Done, thank you!

-----

## Regression Notes

1. Potential unintended areas of impact

    - site switcher

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - manual

8. What automated tests I added (or what prevented me from doing so)

    - n/a

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):

- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -1,5 +1,9 @@
 *** PLEASE FOLLOW THIS FORMAT: [<priority indicator, more stars = higher priority>] <description> [<PR URL>]
 
+25.0
+-----
+* [*] Site picker: Fixed the UI alignment issue in RTL [https://github.com/wordpress-mobile/WordPress-Android/pull/20804]
```
Comment (antonis) [inline RELEASE-NOTES.txt]: Due to timezones I think we have a few more hours [till the release 24.9 is cut](https://github.com/wordpress-mobile/WordPress-Android/tree/release/24.9). Wdyt of targeting 24.9 and updating the release note?

**Review by antonis — APPROVED** [review]
Thank you for the quick fix @jarvislin 🙇 
The code changes LGTM and the app behaves as expected in RTL 🎉 

<details>
<summary>screenshot</summary>

Screenshot_20240513_120128

</details>

File: RELEASE-NOTES.txt (hunk 1 above)
Comment (jarvislin) [reply RELEASE-NOTES.txt]: Sounds good!

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-Android/milestone/284">24.9</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Pull request merged**